src/index.ts

```diff
@@ -37,10 +56,159 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
   }
 }
 
-export default function twiddleEject(twiddleUrl: string, projectName: string, _options: {} = {}) {
+/**
+ * Unzip and extract from a given url and save the data to a path provided.
+ * @param gistUrl
+ * @param filePath
+ * @param targetDir
+ */
+async function unzipAndExtractFromUrl(gistUrl: string, filePath: string, targetDir: string): Promise<string> {
+  const { url: gistLink } = await fetch(gistUrl);
+  const gistDataUrl = `${gistLink}/${filePath}`;
+  const newZip = new AdmZip();
+  const gistResponse = await fetch(gistDataUrl);
+  const gistData = Buffer.from(await gistResponse.arrayBuffer());
+  const zip = new AdmZip(gistData);
+  const zipEntries = zip.getEntries();
+
+  zipEntries.forEach(zipEntry => {
+    const fileName = zipEntry.entryName;
+    const fileContent = Buffer.from(zip.readAsText(fileName), 'utf-8');
+    // Here remove the top level directory
+    const newFileName = fileName.substring(fileName.indexOf('/') + 1);
+    newZip.addFile(newFileName, fileContent, '');
+  });
+
+  // Generate the project root directory.
+  mkdirsSync(targetDir);
+  newZip.extractAllTo(targetDir);
+  log(`Files extracted to path: ${targetDir}`);
+  return targetDir;
+}
+
+/**
+ * Returns the json object for a given path
+ * @param targetDir
+ * @param fileName
+ */
+function getJsonFileData(targetDir: string, fileName: string) {
+  // const file = `${targetDir}/${fileName}.json`;
+  const file = path.join(targetDir, `${fileName}.json`);
+  try {
+    return readJSONSync(file);
+  } catch (err) {
+    throw err;
+  }
+}
+
+/**
+ * Normalize the generated twiddle to a ember-like structure.
+ * @param targetDir
+ */
+function normalizeTwiddle(targetDir: string) {
+  const appDir = path.join(targetDir, 'app');
+  mkdirsSync(appDir);
+  const files = fs.readdirSync(targetDir);
+  for (let i in files) {
+    if (files.hasOwnProperty(i)) {
+      const file = files[i];
+      const dirPath = file.split('.');
+      const fileName = dirPath.splice(-2, 2);
+      const newPath = path.join(appDir, ...dirPath);
+      const filePath = path.join(newPath, fileName.join('.'));
```

Comment:
I'd break this path manipulation out into its own function so we can unit test it

```js
const file = files[i];
const filePath = getFilePath(file); // < extract the splitting, splicing, joining into a new function
```